docs/source/add_new_model.rst

```diff
@@ -0,0 +1,811 @@
+-  ☐ Created script that successfully runs forward pass using original repository and checkpoint
+-  ☐ Successfully added the model skeleton to Transformers
+-  ☐ Successfully converted original checkpoint to Transformers checkpoint
+-  ☐ Successfully ran forward pass in Transformers that gives identical output to original checkpoint
+-  ☐ Finished model tests in Transformers
+-  ☐ Successfully added Tokenizer in Transformers
+-  ☐ Run end-to-end integration tests on GPU
+-  ☐ Finished docs
+-  ☐ Uploaded model weights to the hub
+-  ☐ Merged the pull request
+-  ☐ (Optionally) added a notebook
+
+For exemplary purposes, we will call the model that we will add to 🤗 Transformers BrandNewBert. To begin with, you
+should start by getting a good understanding of BrandNewBert.
+
+Theoretical aspects of BrandNewBert
+-----------------------------------------------------------------------------------------------------------------------
+
+You should take some time to read *brand_new_bert's* paper. There might be large sections of the paper that are
+difficult to understand. If this is the case, this is fine - don’t worry! The goal is not to get a deep theoretical
+understanding of the paper, but to extract the necessary information required to effectively re-implement the model to
+🤗 Transformers. That being said, you don’t have to spend too much time on the theoretical aspects, but rather focus on
+the practical ones, namely:
+
+-  What type of model is *brand_new_bert*? BERT-like encoder-only model? GPT2-like decoder-only model? BART-like
+   encoder-decoder model?
+-  What are the applications of *brand_new_bert*? Text classification? Text generation? Seq2Seq tasks, *e.g.,*
+   summarization?
+-  What is the novel feature of the model making it different from BERT or BART (if it’s an encoder-decoder model)?
+-  Which of the already existing `🤗 Transformers models <https://huggingface.co/transformers/#contents>`__ is most
+   similar to *brand_new_bert*?
+-  What type of tokenizer is used? A sentencepiece tokenizer? Word piece tokenizer? Is it the same tokenizer as used
+   for BERT or BART?
+
+After you feel like you have gotten a good overview of the architecture of the model, you might want to write the
+Hugging Face team for any questions you might have. This might include questions regarding the model’s architecture,
+its attention layer, etc. We will be more than happy to help you.
+
+Next prepare your environment
+-----------------------------------------------------------------------------------------------------------------------
+
+1. Fork the `repository <https://github.com/huggingface/transformers>`__ by clicking on the ‘Fork’ button on the
+   repository’s page. This creates a copy of the code under your GitHub user account.
+
+2. Clone your fork to your local disk, and add the base repository as a remote:
+
+   .. code:: bash
+
+      git clone https://github.com/[your Github handle]/transformers.git
```

Comment:
I would prefer using `git@github.com:[your Github handle]/transformers.git`, because when using the clone over https you're required to enter your GitHub user name + password, whereas this solution requires a proper SSH key setup.

(Btw: https clone don't work when you - hopefully - use 2-factor auth with GitHub)

Reply:
or they can use my script that does all the setup work for you (over ssh though):
https://github.com/stas00/git-tools/blob/master/how-to-make-pr/github-make-pr-branch
and then I just have aliases:
```
alias trbr="github-make-pr-branch ssh stas00 huggingface transformers"
alias dabr="github-make-pr-branch ssh stas00 huggingface datasets"
alias blbr="github-make-pr-branch ssh stas00 huggingface blog"
alias dsbr="github-make-pr-branch ssh stas00 microsoft DeepSpeed"
alias ptbr="github-make-pr-branch ssh stas00 pytorch pytorch"
alias fsbr="github-make-pr-branch ssh stas00 facebookresearch fairscale"
```